nightfall-client/src/services/commitment-storage.mjs

```diff
@@ -93,11 +135,14 @@ async function findUsableCommitments(zkpPublicKey, ercAddress, tokenId, _value,
       'preimage.ercAddress': ercAddress.hex(32),
       'preimage.tokenId': tokenId.hex(32),
       isNullified: false,
+      isPendingNullification: false,
     })
     .toArray();
   if (commitmentArray === []) return null;
   // turn the commitments into real commitment objects
-  const commitments = commitmentArray.map(ct => new Commitment(ct.preimage));
+  const commitments = commitmentArray
+    .filter(commitment => Number(commitment.isOnChain) > Number(-1)) // filters for on chain commitments
+    .map(ct => new Commitment(ct.preimage));
 
   // Now filter all commitments to also work with those that timber has already seen.
   const knownCommitments = (
```

Comment:
I believe this line is now redundant because the `.filter(commitment => Number(commitment.isOnChain) > Number(-1))` has the same effect of checking that `commitment.index !== null` . The variable names will just need to be updated.

I think this had to be previously split up into two calls because promisfying `.filter` does work/ is not composable

Reply:
I missed that - makes sense! Will revert that change, thanks